File: apps/ios/Sources/Srs.swift
import Foundation

enum Srs {
    static let dayMs: Int = 86_400_000
    static let incorrectDelayMs: Int = 120_000

    static func nextIntervalDays(previous: Int, isCorrect: Bool) -> Int {
        if !isCorrect { return 0 }
        let doubled = previous == 0 ? 1 : previous * 2
        return max(1, doubled)
    }

    static func dueOffsetMs(previous: Int, isCorrect: Bool) -> Int {
        if !isCorrect { return incorrectDelayMs }
        let interval = nextIntervalDays(previous: previous, isCorrect: isCorrect)
        return interval * dayMs
    }
}

File: apps/android/app/src/main/java/com/learnjapanese/app/data/Srs.kt
package com.learnjapanese.app.data

object Srs {
    const val DAY_MS: Long = 86_400_000L
    const val INCORRECT_DELAY_MS: Long = 120_000L

    fun nextIntervalDays(
        previous: Int,
        isCorrect: Boolean,
    ): Int {
        if (!isCorrect) return 0
        val doubled = if (previous == 0) 1 else previous * 2
        return maxOf(1, doubled)
    }

    fun dueOffsetMs(
        previous: Int,
        isCorrect: Boolean,
    ): Long {
        if (!isCorrect) return INCORRECT_DELAY_MS
        val interval = nextIntervalDays(previous, isCorrect)
        return interval * DAY_MS
勿    }
}
